### PR TITLE
ISSUE-174 Fix _json_key GCR authentication

### PR DESCRIPTION
**Pull request by ivanilves**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,11 +30,9 @@ script:
   - make lint
   - make vet
   - if [[ "${TRAVIS_PULL_REQUEST}" != "false" ]]; then make semantic RANGE=$(scripts/travis_range.sh); fi
-  - make stress-test-async CONCURRENT_REQUESTS=256
```
Comment (vonrabbe) [inline .travis.yml]: 😞 

@ivanilves maybe move to CircleCI? Travis is such a piece of crap these days...

File: docker/config/config.go [hunk 2]
```diff
@@ -38,24 +38,44 @@ func (c *Config) IsEmpty() bool {
...
 	if _, defined := c.usernames[registry]; !defined {
-		username, password, _ := credhelper.GetCredentials(registry, c.CredsStore, c.CredHelpers)
+		username, password, err := credhelper.GetCredentials(
+			registry,
+			c.CredsStore,
+			c.CredHelpers,
+		)
 
-		return username, password, false
+		if err != nil {
```
Comment (vonrabbe) [inline docker/config/config.go]: LOL! How it was working before?!

File: docker/config/config.go [hunk 3]
```diff
@@ -38,24 +38,44 @@ func (c *Config) IsEmpty() bool {
...
+
+		return username, password, true
 	}
 
 	return c.usernames[registry], c.passwords[registry], true
 }
 
+func getAuthJSONString(username, password string) string {
+	if username == "_json_key" {
+		return fmt.Sprintf("%s:%s", username, password)
```
Comment (vonrabbe) [inline docker/config/config.go]: Weird but OK as official Docker CLI does the same ..

File: .travis.yml (hunk 1 above)
Comment (ivanilves) [reply .travis.yml]: I'll definitely think about this. I don't depend on any Travis-specific things.

File: docker/config/config.go (hunk 2 above)
Comment (ivanilves) [reply docker/config/config.go]: Don't ask! 😛

File: docker/config/config.go (hunk 3 above)
Comment (ivanilves) [reply docker/config/config.go]: Weird indeed but this is how the things work.

**Pull request merged**